wallet-test/src/test/scala/org/bitcoins/wallet/UTXOLifeCycleTest.scala

```diff
@@ -0,0 +1,73 @@
+package org.bitcoins.wallet
+
+import org.bitcoins.core.currency.Satoshis
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.protocol.script.EmptyScriptPubKey
+import org.bitcoins.core.protocol.transaction.TransactionOutput
+import org.bitcoins.core.wallet.fee.SatoshisPerVirtualByte
+import org.bitcoins.core.wallet.utxo.TxoState
+import org.bitcoins.testkit.wallet.BitcoinSWalletTest
+import org.bitcoins.testkit.wallet.BitcoinSWalletTest.WalletWithBitcoind
+import org.scalatest.FutureOutcome
+
+class UTXOLifeCycleTest extends BitcoinSWalletTest {
+
+  behavior of "Wallet Txo States"
+
+  override type FixtureParam = WalletWithBitcoind
+
+  val testAddr: BitcoinAddress =
+    BitcoinAddress.fromString("n4MN27Lk7Yh3pwfjCiAbRXtRVjs4Uk67fG").get
+
+  override def withFixture(test: OneArgAsyncTest): FutureOutcome = {
+    withFundedWalletAndBitcoind(test)
+  }
+
+  it should "track a utxo state change to pending spent" in { param =>
+    val WalletWithBitcoind(wallet, _) = param
+
+    for {
```

Comment:
nit: So one problem I have with these tests is we aren't being specific enough. What utxo are we spending? What is the initial state (I'm guessing `ConfirmedReceived`)? These should both be asserted on. Maybe this is just being too pedantic about testing, so I'll let you be the judge if you want to address this or not.

Reply:
I had this originally, but think it is better as it is now because then we don't rely on `withFundedWalletAndBitcoind` to set it up so the txs have the correct confirmations as well as, these tests are just making sure it becomes the correct state, I think it makes more sense for it to be previous state agnostic